storm-core/src/jvm/backtype/storm/scheduler/TopologyDetails.java

```diff
@@ -396,34 +410,50 @@ public void addResourcesForExec(ExecutorDetails exec, Map<String, Double> resour
             LOG.warn("Executor {} already exists...ResourceList: {}", exec, getTaskResourceReqList(exec));
             return;
         }
-        _resourceList.put(exec, resourceList);
+        this.resourceList.put(exec, resourceList);
     }
 
     /**
      * Add default resource requirements for a executor
      */
     public void addDefaultResforExec(ExecutorDetails exec) {
+        Double topologyComponentCpuPcorePercent = Utils.getDouble(this.topologyConf.get(Config.TOPOLOGY_COMPONENT_CPU_PCORE_PERCENT), null);
+        if (topologyComponentCpuPcorePercent == null) {
+            LOG.warn("default value for " + Config.TOPOLOGY_COMPONENT_CPU_PCORE_PERCENT + " needs to be set!");
+        }
```

Comment:
@jerrypeng, if we print a message in the unit test similar to the one we have now, "some.config.key is missing from defaults.yaml", would this be enough to address your concern about how to find the cause of the test failure?